hathor/p2p/downloader.py

```diff
@@ -127,6 +144,20 @@ def __init__(self, manager: 'HathorManager', window_size: int = 100):
         self.downloading_buffer = {}
         self.window_size = window_size
 
+    def drop_connection(self, connection: 'NodeSyncTimestamp') -> None:
+        """ Called when a peer is disconnected to remove that connection from all the affected pending transactions."""
+        to_remove: List[bytes] = []
+        for tx_details in self.pending_transactions.values():
+            tx_details.drop_connection(connection)
+            if not tx_details.has_available_connection():
+                to_remove.append(tx_details.tx_id)
+                if tx_details.downloading_deferred is not None:
```

Comment:
Should we move this whole logic to the `tx_details.drop_connection(...)`?

Reply:
Yeah, that makes sense.